Source/CarthageKit/Project.swift

```diff
@@ -189,31 +189,20 @@ public final class Project {
 				return false
 			}
 		}
-
-		let cartfile = SignalProducer.attempt {
-				return Cartfile.fromFile(cartfileURL)
-			}
-			.flatMapError { error -> SignalProducer<Cartfile, CarthageError> in
-				if isNoSuchFileError(error) && NSFileManager.defaultManager().fileExistsAtPath(privateCartfileURL.path!) {
```

Comment:
Loading `Cartfile` should be error out if both `Cartfile` and `Cartfile.private` dont't exist, and the error is successfully ignored in [`buildProjectInDirectoryURL` if `--no-skip-current` option is specified](https://github.com/Carthage/Carthage/blob/0.17/Source/carthage/Build.swift#L143-L149).

Thus removing the latter condition breaks the option appeared in #1366.